### ISSUE TITLE
unexpected token error in /httpntlm/ntlm.js:427

### ISSUE THREAD

**Issue by Nishikant26** [SamDecrock/node-http-ntlm#102]
var timestamp = ((BigInt(now) + 11644473600000n) * 10000n);
                                                    ^^^^^^^^^^^^
SyntaxError: Invalid or unexpected token

Since today morning, we are getting the above error while loading the npm server.
This issue occurred in 1.8.0 which was published 2 days ago.

Please fix the issue ASAP.

**Comment by SamDecrock** [comment]
You can fix this by specifying the exact version in your `package.json`.

```json
"httpntlm": "1.7.7"
```

In the mean while, I'll have a look at the issue.

**Comment by totocald** [comment]
@SamDecrock, don't you think it's better to protect your code by converting a literal value to BigInt? As in the example.
`// 11644473600000 = diff between 1970 and 1601
	var now = Date.now();
	var timestamp = ((BigInt(now) + BigInt(11644473600000)) * BigInt(10000));`

**Comment by SamDecrock** [comment]
I can do that if it makes you feel better. What's the difference if I may ask?

**Comment by totocald** [comment]
Thank you very much for your reply.

Regarding your question, I think it would benefit us to check compatibility on background, as we sometimes use one or more libraries that have httpntlm as a co-dependency. These libraries use dynamic versions in their dependencies(^,~, etc.).
So sometimes it's harder to trace or harden at the dependency level.
It may seem silly, but it could help us have fewer problems when using this library.

I really appreciate your attention.

**Comment by vit100** [comment]
Indeed, @totocald is right. It is better to have lib with defensive code. 

For example **soap** package is using httpntlm, and in node 8 it stopped working recently, after new httpntlm release. I agree, work around will be to put "httpntlm": "1.7.7" into dependencies. But it is kind of hack
